src/library/attribute_manager.cpp

```diff
@@ -81,6 +82,10 @@ struct attr_record {
     bool deleted() const {
         return !static_cast<bool>(m_data);
     }
+
+private:
+    /* no not use `m_persistent` in comparison */
+    bool operator == (attr_record const & other);
```

Comment:
```suggestion
    bool operator == (attr_record const & other) = 0;
```

The equality operator is not used anywhere, right?

Reply:
Ah, I didn't test this suggestion.  The ` = 0` allows you to disable auto-generated functions in C++.  But the `operator==` is not auto-generated, you can just remove this line completely.